test/dynamo/test_aot_autograd_cache.py

```diff
@@ -1,19 +1,270 @@
 # Owner(s): ["module: dynamo"]
 
+import os
+import unittest
+
 import torch
 import torch._dynamo
 import torch._dynamo.test_case
 
 import torch._functorch._aot_autograd
+from torch._dynamo.utils import counters
 from torch._functorch import config as functorch_config
 from torch._functorch._aot_autograd.autograd_cache import (
-    autograd_cache_hash,
+    AOTAutogradCache,
+    autograd_cache_key,
     BypassAOTAutogradCache,
 )
 from torch._functorch._aot_autograd.schemas import AOTConfig
 from torch._inductor import config as inductor_config
-
-
+from torch.testing._internal.common_cuda import SM80OrLater
+from torch.testing._internal.common_device_type import largeTensorTest
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+)
+from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_GPU
+
+
+@instantiate_parametrized_tests
+class AOTAutogradCacheTests(torch._dynamo.test_case.TestCase):
```

Comment:
Thinking some more about testing - we probably want some way to feel confident that our caching logic properly serializes and runs the same set of runtime boundary conditions that happen on a cache miss.

`test/functorch/test_aotdispatch.py::TestAOTAutograd` ([file](https://github.com/pytorch/pytorch/blob/main/test/functorch/test_aotdispatch.py#L296)) is a nice trove of tests for various boundary conditions (e.g. usage of `indices_of_inps_to_detach`). Unfortunately, they all are written to use a helper function `verify_aot_autograd`, which uses `aot_function/aot_module` ([link](https://github.com/pytorch/pytorch/blob/main/test/functorch/test_aotdispatch.py#L364)), so none of them will exercise caching.

Maybe it's not worth the squeeze, but what do you think about getting those tests to all exercise the warm cache? One option would be something like:

(1) invoke that entire test suite, but with a flag set (`test_warm_cache=True`)

(2) when that flag is set, `verify_aot_autograd` should just use `torch.compile(f, backend="aot_eager")`, instead of manually using AOTAutograd's other entrypoints - and then run twice to see if we get the same results on a cache hit.

Reply:
Yes I think that'd be great: I'll start working on this at the top of my stack